module/VuFind/tests/unit-tests/src/VuFindTest/Auth/LoginTokenTest.php

```diff
@@ -0,0 +1,249 @@
+<?php
+
+/**
+ * Class AuthTokenTest
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library of Finland 2023
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  VuFindTest\Auth
+ * @author   Jaro Ravila <jaro.ravila@helsinki.fi>
+ * @license  https://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+declare(strict_types=1);
+
+namespace VuFindTest\Auth;
+
+use VuFind\Auth\LoginToken;
+use Laminas\Config\Config;
+use VuFind\Db\Row\User as UserRow;
+use VuFind\Db\Table\User as UserTable;
+use VuFind\Db\Row\LoginToken as LoginTokenRow;
+use Laminas\Session\SessionManager;
+use VuFind\Cookie\CookieManager;
+use VuFind\Exception\LoginToken as LoginTokenException;
+
+
+/**
+ * Class AuthTokenTest
+ *
+ * @category VuFind
+ * @package  VuFindTest\Auth
+ * @author   Jaro Ravila <jaro.ravila@helsinki.fi>
+ * @license  https://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+class LoginTokenTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * Test logging in with a login token
+     *
+     * @return void
+     */
+    public function testTokenLogin()
+    {
+        $user = $this->getMockUser();
+        $cookieManager = $this->getCookieManager(
+            [
+              'loginToken' => '222;0;111'
+            ]
+        );
+        $mockToken = $this->getMockLoginToken();
+        $userTable = $this->getMockUserTable();
+        $userTable->expects($this->once())->method('getById')
+            ->with($this->equalTo(0))
+            ->will($this->returnValue($this->getMockUser()));
+        $tokenTable = $this->getMockLoginTokenTable();
+        $tokenTable->expects($this->once())->method('matchToken')
+            ->will($this->returnValue($mockToken));
+        $loginToken = $this->getLoginToken($cookieManager, $tokenTable, $userTable);
+
+        // Expect exception due to browscap.ini requirements
+        $this->expectException(\VuFind\Exception\Auth::class);
```

Comment:
Should the name of this test method indicate that we're expecting a failure/exception? The name suggests that this is testing a normal successful run, but that doesn't appear to be the case.